tools/importers/SynchronizeTypographyTokens.ts

```diff
@@ -0,0 +1,191 @@
+						label: propertyName,
+						value: propertyValue,
+					};
+
+					if (resultJsonCss[key] === undefined) {
+						resultJsonCss[key] = [];
+					}
+					resultJsonCss[key].push(resultJsonObject);
+				}
+			}
+		}
+	}
+
+	resultCssAdditionalLines.forEach((item) => {
+		resultCss.push(item);
+	});
+	resultScssAdditionalLines.forEach((item) => {
+		resultScss.push(item);
+	});
+
+	arrayToFile(
+		tokensFilesConfig.destinationPath + binFilesConfig.variablesRaw.destinationVariablesCss,
+		resultCss,
+	);
+	arrayToFile(
+		tokensFilesConfig.destinationPath + binFilesConfig.variablesRaw.destinationVariables,
+		resultScss,
+	);
+	jsonToFile(
+		tokensFilesConfig.destinationPath + binFilesConfig.variablesRaw.destinationVariablesCssJson,
+		resultJsonCss,
+	);
+};
+
+const ImportTypographyTokens = (binConfig: TypographyBinFiles, jsonTypography: any) => {
+	let resultScss: Array<ITypographyToken> = recursiveTokensReader(jsonTypography[tokensKey], '');
+	let resultJsonCss: Dict<Array<ITypographyToken>> = { tokens: resultScss };
+
+	arrayToMixinFile(tokensFilesConfig.destinationPath + binConfig.tokens.destination, [
+		importVariables,
+		...buildTypographyTokensMixins(resultScss.flat()),
+	]);
+	jsonToFile(tokensFilesConfig.destinationPath + binConfig.tokens.destinationJson, resultJsonCss);
+};
+
+function buildTypographyTokensMixins(tokens: Array<ITypographyToken>): Array<string> {
+	let result: Array<string> = [];
+
+	for (let key in tokens) {
+		result.push(mixinNameLine(tokens[key].tokenCamelCase));
+		result.push(tokenAsCssPropertyLine(tokens[key].token));
+		tokens[key].attributes.forEach((attribute) => {
+			let [cssAttributePartOne, cssAttributePartTwo] = attribute.split('-');
+			result.push(
+				cssAttributePartOne +
+					'-' +
+					cssAttributePartTwo +
+					': $' +
+					typographyPrefix +
+					attribute +
+					';',
+			);
+		});
+		result.push(closeBracketLine());
+	}
+
+	return result;
+}
+
+const SynchronizeTypographyTokensBin = async () => {
+	console.log('Import in progress...');
+	let requestResponse = await requestForBin(tokensFilesConfig.bin);
+
+	ImportTypographyVariables(
+		tokensFilesConfig.bin.files,
+		requestResponse.data.record.values.LMSDesignSystemTypography,
+	);
+
+	ImportTypographyTokens(
+		tokensFilesConfig.bin.files,
+		requestResponse.data.record.values.LMSDesignSystemTypography,
+	);
+
+	console.log('The import was successful for bin: ' + tokensFilesConfig.bin.id);
+};
+
+const requestForBin = async (binConfig: TypographyConfigFileBin) => {
+	const requestConfig = {
+		headers: {
+			'X-Access-Key': process.env.JSON_BIN_X_MASTER_KEY,
+		},
+	};
+	const requestResponse = await axios.get(
+		tokensFilesConfig.jsonBinApiUrl + binConfig.id + '/latest',
+		requestConfig,
+	);
+
+	if (!requestResponse.data.record.values.LMSDesignSystemTypography) {
+		throw new TypeError('Response structure has no content!');
+	}
+
+	return requestResponse;
+};
+
+SynchronizeTypographyTokensBin().finally(() => {
```

Comment:
there is a lot of logic in this importer. Maybe we should cover it with tests

Reply:
Testes for typography synchronization added. Please check used convention. I will make tests for colors synchronization only if these will be acceptable.